git-flow: initial draft commit

alex-pezzati opened the pull request:
- initial git flow guide draft with images
- setup for future expansion, if needed

chaunceykm commented on git-flow-guide/README.md:
```diff
@@ -0,0 +1,172 @@
+
+### What is the recommended git flow for long project problems?
+
+-   For longer practices it is recommended that pairs follow a/A suggested git flow of working from the same repo while taking turns swapping roles between driving and navigating.
```
I would change recommended to expected. I don't want students to believe that pairing is an optional practice for afternoon practices.

alex-pezzati replied:
Made the requested changes. Thanks for the feedback!

chaunceykm reviewed (changes requested):
minor correction needed

barrymattern commented:
Alex, as I mentioned, this is coming along quite nicely – having this info all in one place/repo will be very valuable for the students. As I read through these were some things that came to mind. Let me know if you'd like to talk through any of them  👍

# Thoughts

- For step-by-step guides an idea I've been taught is to write your draft, then cut your text in half  ...and then cut it in half again
  - Goal is to get the steps at a quick glance
  - Then expand to a secondary or longer guide (or second page) to go more in-depth
- Students will more than likely be in a rush when reading this
  - Might be good to have 2 versions of this guide:
    1. Short/quick, step by step guide for students to follow in the moment
    2. Long/in-depth guide for TAs to update and review as needed
- There's some great info in here, and a nice layout
  - I'm wondering if these should be broken out into separate guides, then all connected in one main guide?
  - For example, initial README could have links to each quick secondary guide (like an index)

---

### Quick idea for student-facing guide to update remote URL:

1. Double check remote URL
```
git remote -v
``` 
2. If it's a/A's URL, update to your repo's URL
```
git remote set-url origin https://github.com/USERNAME/REPOSITORY.git
```
  - When done correctly it looks like nothing happens
3. Double check remote URL one more time
```
git remote -v
``` 
  - You may need to add and commit a change in order to push

These commands & more can be found in the [GitHub Docs](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)

alex-pezzati commented:
> Alex, as I mentioned, this is coming along quite nicely – having this info all in one place/repo will be very valuable for the students. As I read through these were some things that came to mind. Let me know if you'd like to talk through any of them 👍
> 
> # Thoughts
> * For step-by-step guides an idea I've been taught is to write your draft, then cut your text in half  ...and then cut it in half again
>   
>   * Goal is to get the steps at a quick glance
>   * Then expand to a secondary or longer guide (or second page) to go more in-depth
> * Students will more than likely be in a rush when reading this
>   
>   * Might be good to have 2 versions of this guide:
>     
>     1. Short/quick, step by step guide for students to follow in the moment
>     2. Long/in-depth guide for TAs to update and review as needed
> * There's some great info in here, and a nice layout
>   
>   * I'm wondering if these should be broken out into separate guides, then all connected in one main guide?
>   * For example, initial README could have links to each quick secondary guide (like an index)
> 
> ### Quick idea for student-facing guide to update remote URL:
> 1. Double check remote URL
> 
> ```
> git remote -v
> ```
> 
> 1. If it's a/A's URL, update to your repo's URL
> 
> ```
> git remote set-url origin https://github.com/USERNAME/REPOSITORY.git
> ```
> 
> * When done correctly it looks like nothing happens
> 
> 1. Double check remote URL one more time
> 
> ```
> git remote -v
> ```
> 
> * You may need to add and commit a change in order to push
> 
> These commands & more can be found in the [GitHub Docs](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)

I was thinking I could circumvent the need for multiple guides with just efficient navigation links. I could add a "back to top of section" link as well. Just to move around between content effortlessly.

I do agree it is probably too wordy right now. Let me see what I can reduce it down to.

chaunceykm commented on git-flow-guide/README.md:
```diff
@@ -15,7 +15,7 @@ TO DO:
 # Git Flow
 
-This is a guide with suggestions for _recommended_ practices for git and GitHub while in a/A's Module 2: Computer Science and continued studies at a/A. Let's git into it!
+This is a guide with for _expected_ practices for git and GitHub while in a/A's Module 2: Computer Science and continued studies at a/A. Let's git into it!
```
I think this needs a look over. The phrasing seems a bit off.

alex-pezzati replied:
Good catch. My fix before was too quick. Pushed. Moving onto reducing some of the bloat next.

alex-pezzati commented:
Alright, I will make another pass on Monday although I already made a bunch of changes:

- reduced the word count in many places, 
- separated paragraphs/sentences into actionable bullet points, 
- added code blocks for easy copy/pasting, 
- added external links section, 
- added more comments as documentation for further expansion


This is seems like a pretty good state for this next cohort at least.
